src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -93,6 +96,50 @@
   (instance? X509CRL x))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def ValidX500Name
+  (schema/pred valid-x500-name?))
+
+(def Extension
+  "A map containing all the fields required to define an extension."
+  {:oid schema/Str
+   :critical (schema/pred #(not (nil? %)))
```

Comment:
I'd just been thinking of using `Boolean` here instead of the `(schema/pred ...)`.  Can't remember if there are any cases where we'd expect this to be passed around as a string but converted to a boolean later...